avogadro/qtplugins/openbabel/obcharges.cpp

```diff
@@ -198,6 +198,10 @@ MatrixX OBCharges::partialCharges(Core::Molecule& molecule) const
     charges(i, 0) = output[i];
   }
 
+  // workaround failed runs causing the code to freeze
+  if (abs(charges(0, 0)) < 0.00001)
```

Comment:
This part, I'm not sure I understand.

Reply:
Yes, exactly. In that case it would just freeze, this gets it to finish as soon as possible.

Reply:
I'm sure there are better ways to do it, but this one is quite simple.

Reply:
Okay, I'll merge for now while I think about a better solution.